packages/cysync-core/src/dialogs/Receive/context/index.tsx

```diff
@@ -0,0 +1,147 @@
+  useMemo,
+  useState,
+} from 'react';
+
+import { selectLanguage, useAppSelector } from '~/store';
+
+import { Receive, ReceiveDevice } from '../Dialogs';
+import {
+  ReceiveDeviceConfirm,
+  ReceiveDeviceConfirmCancelled,
+  ReceiveDeviceConfirmForToken,
+  ReceiveDeviceConfirmTroubleShoot,
+  ReceiveDeviceConnection,
+} from '../Dialogs/Device';
+import {
+  ReceiveAddressNotVerified,
+  ReceiveAddressVerified,
+  ReceiveVerifyAddress,
+} from '../Dialogs/Receive';
+
+type ITabs = {
+  name: string;
+  dialogs: ReactNode[];
+}[];
+
+export interface ReceiveDialogContextInterface {
+  tabs: ITabs;
+  currentTab: number;
+  setCurrentTab: Dispatch<SetStateAction<number>>;
+  currentDialog: number;
+  setCurrentDialog: Dispatch<SetStateAction<number>>;
+  onNext: (tab?: number, dialog?: number) => void;
+  onPrevious: () => void;
+}
+
+export const ReceiveDialogContext: Context<ReceiveDialogContextInterface> =
+  createContext<ReceiveDialogContextInterface>(
+    {} as ReceiveDialogContextInterface,
+  );
+
+export interface ReceiveDialogContextProviderProps {
+  children: ReactNode;
+}
+
+export const ReceiveDialogProvider: FC<ReceiveDialogContextProviderProps> = ({
+  children,
+}) => {
+  const lang = useAppSelector(selectLanguage);
+  const [currentTab, setCurrentTab] = useState<number>(0);
+  const [currentDialog, setCurrentDialog] = useState<number>(0);
+
+  const tabs: ITabs = [
+    {
+      name: lang.strings.receive.aside.tabs.source,
+      dialogs: [<Receive />],
+    },
+    {
+      name: lang.strings.receive.aside.tabs.device,
+      dialogs: [
+        <ReceiveDevice />,
+        <ReceiveDeviceConnection />,
+        <ReceiveDeviceConfirmCancelled />,
+        <ReceiveDeviceConfirmTroubleShoot />,
+        <ReceiveDeviceConfirmForToken />,
+        <ReceiveDeviceConfirm />,
+      ],
+    },
+    {
+      name: lang.strings.receive.aside.tabs.receive,
+      dialogs: [
+        <ReceiveVerifyAddress />,
+        <ReceiveAddressNotVerified />,
+        <ReceiveAddressVerified />,
+      ],
+    },
+  ];
+
+  const onNext = (tab?: number, dialog?: number) => {
+    if (typeof tab === 'number' && typeof dialog === 'number') {
+      setCurrentTab(tab);
+      setCurrentDialog(dialog);
+    } else if (currentDialog + 1 > tabs[currentTab].dialogs.length - 1) {
+      setCurrentTab(prevProps => Math.min(tabs.length - 1, prevProps + 1));
+      if (currentTab !== tabs.length - 1) {
+        setCurrentDialog(0);
+      }
+    } else {
+      setCurrentDialog(prevProps =>
+        Math.min(tabs[currentTab].dialogs.length - 1, prevProps + 1),
+      );
+    }
+  };
+
+  const onPrevious = () => {
+    if (currentDialog - 1 < 0) {
+      if (currentTab === 0) {
+        setCurrentDialog(0);
+      } else {
+        setCurrentDialog(tabs[currentTab - 1].dialogs.length - 1);
+        setCurrentTab(prevProps => Math.max(0, prevProps - 1));
+      }
+    } else {
+      setCurrentDialog(prevProps => Math.max(0, prevProps - 1));
+    }
+  };
```

Comment:
Since these functions are states are present in all dialog boxes, it's better if we create a reusable hook for these states and functions instead of redefining everytime.

Reply:
created